src/edu/csus/ecs/pc2/clics/API202306/CLICSAccount.java

```diff
@@ -0,0 +1,320 @@
+     * @return an Account array of accounts converted from the CLICS accounts or null if error
+     */
+    private static Account [] createAccountsFromCLICSAccounts(IInternalContest contest, CLICSAccount [] accts, int site, Log log) {
+
+        Account [] newaccounts = null;
+        ArrayList<Account> accounts = new ArrayList<Account>();
+        Account account;
+        JSONTool jsontool = new JSONTool(contest, null);
+        PermissionList teamPermissionList = new PermissionGroup().getPermissionList (ClientType.Type.TEAM);
+        boolean error = false;
+
+        // convert each clics account to a pc2 account, first looking up the existing account, if one exists
+        for(CLICSAccount acct: accts) {
+            if(StringUtilities.isEmpty(acct.id)) {
+                // id is required
+                log.log(Log.SEVERE, "unable to get account id");
+                error = true;
+                break;
+            }
+            if(StringUtilities.isEmpty(acct.username)) {
+                // id is required
+                log.log(Log.SEVERE, "unable to get account username");
+                error = true;
+                break;
+            }
+            if(StringUtilities.isEmpty(acct.type)) {
+                // id is required
+                log.log(Log.SEVERE, "unable to get account type");
+                error = true;
+                break;
+            }
+            Type acctType = Type.UNKNOWN;
+            int acctNum;
+            if(acct.type.equalsIgnoreCase("team")) {
+                acctType = Type.TEAM;
+            } else if(acct.type.equalsIgnoreCase("admin")) {
+                acctType = Type.ADMINISTRATOR;
+            } else if(acct.type.equalsIgnoreCase("judge")) {
+                acctType = Type.JUDGE;
+            } else if(acct.type.equalsIgnoreCase("scoreboard")) {
+                acctType = Type.SCOREBOARD;
+            } else if(acct.type.equalsIgnoreCase("feeder")) {
+                acctType = Type.FEEDER;
+            } else {
+                log.log(Log.SEVERE, "unknown account type " + acct.type);
+                error = true;
+                break;
+            }
+            // break username up into type/number
+            ClientId clientId = getClientIdFromUser(acct.username, contest, log);
+            if(clientId == null) {
+                log.log(Log.SEVERE, "bad username format " + acct.username + "; expecting type###");
+                error = true;
+                break;
+            }
+            if(acctType != clientId.getClientType()) {
+                log.log(Log.SEVERE, "bad account type " + acctType.toString() + " for user " + acct.username);
+                error = true;
+                break;
+            }
+            account = contest.getAccount(clientId);
+            if(account == null) {
+                System.err.println("No account found for id: " + acct.id + " name " + acct.name + " username " + acct.username + " password " + acct.password);
```

Comment:
It seems to me that if this condition is aggregious enough to print out to stderr then it probably also should be logged...